### ISSUE TITLE
[Android/iOS] Crash if not in debug

### ISSUE THREAD

**Issue by ishigamii** [facebook/react-native#23075]
## Environment
```
  React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
      Memory: 133.91 MB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 8.11.2 - ~/.nvm/versions/node/v8.11.2/bin/node
      Yarn: 1.9.4 - /usr/local/bin/yarn
      npm: 5.6.0 - ~/.nvm/versions/node/v8.11.2/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        API Levels: 17, 21, 23, 25, 26, 27
        Build Tools: 23.0.1, 23.0.3, 24.0.1, 25.0.0, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.1, 27.0.3, 28.0.2, 28.0.3
        System Images: android-25 | Google APIs Intel x86 Atom_64
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5056338
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.3 => 16.6.3
      react-native: 0.57.8 => 0.57.8
```
## Description
Regression (I think)
When I create a new project with the latest version of react-native (`0.57.8` at the moment), and then launch the app on my android simulator / phone, I get a red screen or error (or a crash in release).
If I pass the pass in debug mode everything goes perfectly normal.

## package.json (the default one)
```
{
  "name": "test578",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "react": "16.6.3",
    "react-native": "0.57.8"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.51.1",
    "react-test-renderer": "16.6.3"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

## Reproducible Demo
The reproduction is very simple as I just did it creating a new project and opening it.

Here are the steps I did : 

1. `react-native init test578`
2. `cd test578`
3. `react-native run-android`

## Error (firebase crash report)
```
Caused by com.facebook.jni.CppException
Invalid regular expression: unrecognized character after (? (index.android.bundle:462)
com.facebook.react.bridge.queue.NativeRunnable.run
```

## Output 
![screenshot_1547894429](https://user-images.githubusercontent.com/26909067/51425625-09591200-1bdf-11e9-8aad-593eea6cb499.png)

## Simulator info 
![screenshot_1547894475](https://user-images.githubusercontent.com/26909067/51425635-2b529480-1bdf-11e9-8ba9-5d2031e39a73.png)

## Phone info
- Nexus 5X - 8.1.0 (same error)
- One Plus One (same error)

4. `react-native run-ios`

## Output 
**Debug not activated**
![simulator screen shot - iphone x - 2019-01-19 at 11 49 11](https://user-images.githubusercontent.com/26909067/51425762-9355aa80-1be0-11e9-99ed-d85c5eeeec9c.png)

**Debug activated**
![simulator screen shot - iphone x - 2019-01-19 at 11 49 23](https://user-images.githubusercontent.com/26909067/51425763-93ee4100-1be0-11e9-8d34-2bbe6527e7b2.png)

Thanks for reading 👍

**Comment by ishigamii** [comment]
To keep on going with the experimentation I installed the previous version (which is `0.57.7`)

## Reproduce Steps

1.  `react-native init --version="0.57.7" test577``
2. `cd test577``
3. `react-native run-android`

Doing so I got the same bad output as before but I found another project that was using the same version (that I made month ago) and check the differences. I found out this 
**Working project**
```
{
  "name": "test577",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "react": "16.6.1",
    "react-native": "0.57.7"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.50.0",
    "react-test-renderer": "16.6.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

**Not Working ** 
```
{
  "name": "test577b",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "react": "16.6.1",
    "react-native": "0.57.7"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.51.1",
    "react-test-renderer": "16.6.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

Seeing the only difference is the use of `"metro-react-native-babel-preset": "0.51.1",` in the not working project instead of the `"metro-react-native-babel-preset": "0.50.0",` 

So I changed it to that and `yarn install` again. 

Good point is that it corrected my problem for the `0.57.7` project that wasn't working on both iOS and Android.

So I tried the same for the `0.57.8` project but sadly it doesn't solve the case for this version.
Hope it will help find the cause

**Comment by ishigamii** [comment]
Problem was due to a regex negative lookbehinds and javascript does not support them.

The error was caused by this regex. It seems strange that it works perfectly in debug and cause this error in dev mode don't know if it is a bug or just normal so I will just close this issue
